cep-sigstore-predicate.md

```diff
@@ -0,0 +1,72 @@
+# CEP - Standardizing the v1 predicate for sigstore attestations
+
+<table>
+<tr><td> Title </td><td> Standardizing the v1 predicate for sigstore attestations </td>
+<tr><td> Status </td><td> Proposed </td></tr>
+<tr><td> Author(s) </td><td> Wolf Vollprecht &lt;wolf@prefix.dev&gt;</td></tr>
+<tr><td> Created </td><td> Feb 18, 2025 </td></tr>
+<tr><td> Updated </td><td> Feb 18, 2025</td></tr>
+<tr><td> Discussion </td><td>  </td></tr>
+<tr><td> Implementation </td><td>  </td></tr>
+</table>
+
+## Abstract
+
+We want to standardize attestations for the conda ecosystem.
+
+### Sigstore Attestations
+
+Sigstore attestations are cryptographic statements about software artifacts that provide:
+
+- Authenticity: Proof of who created/signed the artifact
+- Integrity: Verification that the artifact hasn't been tampered with
+- Transparency: Public record of signatures in a tamper-evident log
+
+### Key Components
+
+- Predicates: JSON documents containing metadata about the signing event, using the `in-toto` format
+- Signatures: Cryptographic proofs made using ephemeral keys
+- Rekor: A tamper-evident log that stores attestations
+- Fulcio: A certificate authority that issues short-lived certificates
+
+In this document, we want to standardize the sigstore predicate for conda packages. The bundle format to be used for sigstore attestations is the `v0.3` bundle format.
+
+## Specification
+
+The in-toto predicate should contain the following fields:
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v0.1",
+    "subject": [{
+        "name": "file-name-0.0.1-h123456_5.conda",
+        "digest": {"sha256": "..."}, ...
+    }],
+    // Schema URL
+    "predicateType": "https://schemas.conda.org/predicate-v1.json",
+    "predicate": {
+        // Canonical URL of the target channel
+        "targetChannel": "https://prefix.dev/conda-forge",
+    }
+}
+```
+
+The `subject` field is already defined in the in-toto specification and contains the name of the package and its digest.
+For conda packages a SHA256 hash MUST be used.
```

Comment:
Suggest rewording to:
> For conda packages, the digest MUST include the package's SHA256.

The specification should be clear if only the SHA256 can be included, or if optionally, other package checksums can be included as well.

Reply:
Reworded to: 

>     - `subject[0].digest` **MUST** be a [`DigestSet`], and it **MUST** contain
      a single `sha256` entry with the SHA256 hash of the conda package.